packages/main/src/plugin/kubernetes/context-permissions-checker.ts

```diff
@@ -37,46 +37,58 @@ export interface ContextPermissionsRequest {
   onDenyRequests?: ContextPermissionsRequest[];
 }
 
-// ContextResourcePermission is the permission for a specific resource
-export interface ContextResourcePermission {
+// Permission is the permission for a specific resource
+export interface Permission {
   attrs: V1ResourceAttributes;
   // permitted if allowed and not denied
   permitted: boolean;
   reason?: string;
 }
 
+// ContextResourcePermission is the permission for a specific resource in a specific context
+export interface ContextResourcePermission extends Permission {
+  contextName: string;
+  resourceName: string;
+}
+
 // ContextPermissionResult is the result of a request, which can cover several resources
-export interface ContextPermissionResult extends ContextResourcePermission {
+export interface ContextPermissionResult extends Permission {
   kubeConfig: KubeConfigSingleContext;
   resources: string[];
 }
 
 export class ContextPermissionsChecker implements Disposable {
+  #contextName: string;
   #kubeconfig: KubeConfigSingleContext;
   #client: AuthorizationV1Api;
   #request: ContextPermissionsRequest;
   #subCheckers: ContextPermissionsChecker[] = [];
-  #results: Map<string, ContextResourcePermission>;
+  #results: ContextResourcePermission[];
 
   #onPermissionResult = new Emitter<ContextPermissionResult>();
   onPermissionResult: Event<ContextPermissionResult> = this.#onPermissionResult.event;
 
   // builds a ContextPermissionsChecker which will check permissions on the current context of the given kubeConfig
   // The request will be made with `attrs` and if allowed, permissions will be given for `resources`
   // If the result is denied, `onDenyRequests` will be started
-  constructor(kubeconfig: KubeConfigSingleContext, request: ContextPermissionsRequest) {
+  constructor(kubeconfig: KubeConfigSingleContext, contextName: string, request: ContextPermissionsRequest) {
     this.#kubeconfig = kubeconfig;
+    this.#contextName = contextName;
     this.#request = request;
-    this.#results = new Map<string, ContextResourcePermission>();
+    this.#results = [];
     this.#client = this.#kubeconfig.getKubeConfig().makeApiClient(AuthorizationV1Api);
   }
 
+  public isForContext(contextName: string): boolean {
+    return this.#contextName === contextName;
+  }
```

Comment:
hello, I have some remarks about this method.

For example, why the context of this permission checker is not something we can get from the outside ? like `permissionChecker.contextName` (readonly attribute)

b/c the isForContext seems to be for another usecase

or accept ContextHealthState attribute


permissionChecker.matchContextHealthState(ctx): boolean